src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,132 @@
 package core.basesyntax;
 
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
+            throws DataGettingException {
+
+        if (names == null || data == null || dateFrom == null
+                || dateTo == null) {
+            throw new DataGettingException("one or more input arguments are null");
+        }
+
+        int dayFrom = 0;
+        int monthFrom = 0;
+        int yearFrom = 0;
+
+        int dayTo = 0;
+        int monthTo = 0;
+        int yearTo = 0;
+
+        try {
+            dayFrom = getNumberOfDay(dateFrom, "day");
+            monthFrom = getNumberOfDay(dateFrom, "month");
+            yearFrom = getNumberOfDay(dateFrom, "year");
+
+            dayTo = getNumberOfDay(dateTo, "day");
+            monthTo = getNumberOfDay(dateTo, "month");
+            yearTo = getNumberOfDay(dateTo, "year");
+        } catch (WrongWordException e) {
+            System.out.println("Input wrong word");
+        }
+
+        if (yearFrom > yearTo || (yearFrom == yearTo && monthFrom > monthTo)
+                || (yearFrom == yearTo && monthFrom == monthTo && dayFrom > dayTo)) {
+            throw new DataGettingException("Start date cannot be later than end date");
+        }
+
+        String[] startResult = new String[data.length];
+
+        for (int i = 0; i < data.length; i++) {
+            String[] dataArrayI = data[i].split(" ");
+            String[] dataPeriod = dataArrayI[0].split("\\.");
+
+            int day = Integer.parseInt(dataPeriod[0]);
+            int month = Integer.parseInt(dataPeriod[1]);
+            int year = Integer.parseInt(dataPeriod[2]);
+
+            boolean withinDateRange = (year == yearFrom && (month >= monthFrom
+                    && day >= dayFrom) && (month <= monthTo && day <= dayTo))
+                    || (year == yearTo && (month <= monthTo && day <= dayTo)
+                    && (month >= monthFrom && day >= dayFrom))
+                    || (year > yearFrom && year < yearTo);
```

Comment:
The logic for checking if a date is within the specified range is incorrect. Consider using `LocalDate` and its `isBefore` and `isAfter` methods for more accurate date range checks.